sdk/servicebus/azure_messaging_servicebus/src/client.rs

```diff
@@ -0,0 +1,559 @@
+            queue_name.to_string()
+        };
+
+        // Authorize the path if we have a credential-based client
+        self.authorize_path(&entity_path).await?;
+
+        Receiver::new(
+            self.connection.clone(),
+            entity_path,
+            None,
+            options.receive_mode,
+            self.options.clone(),
+        )
+        .await
+    }
+
+    /// Creates a receiver for the specified topic and subscription with options.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// use azure_identity::DeveloperToolsCredential;
+    /// use azure_messaging_servicebus::{ServiceBusClient, CreateReceiverOptions, SubQueue};
+    ///
+    /// #[tokio::main]
+    /// async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    ///     // ... create client ...
+    ///     let credential = DeveloperToolsCredential::new(None)?;
+    ///     let client = ServiceBusClient::builder()
+    ///         .open("my-servicebus.servicebus.windows.net", credential.clone()).await?;
+    ///
+    ///     // Create regular subscription receiver
+    ///     let receiver = client.create_receiver_for_subscription(
+    ///         "my-topic",
+    ///         "my-subscription",
+    ///         None
+    ///     ).await?;
+    ///
+    ///     // Create receiver for subscription's dead letter queue
+    ///     let receiver = client.create_receiver_for_subscription(
+    ///         "my-topic",
+    ///         "my-subscription",
+    ///         Some(CreateReceiverOptions {
+    ///             sub_queue: Some(SubQueue::DeadLetter),
+    ///             ..Default::default()
+    ///         })
+    ///     ).await?;
+    ///     Ok(())
+    /// }
+    /// ```
+    pub async fn create_receiver_for_subscription(
+        &self,
+        topic_name: &str,
+        subscription_name: &str,
+        options: Option<CreateReceiverOptions>,
+    ) -> Result<Receiver> {
+        // For topic subscriptions, the base path is topic/subscriptions/subscription
+        let base_path = format!("{}/subscriptions/{}", topic_name, subscription_name);
+
+        // Build the entity path based on sub_queue option
+        let options = options.unwrap_or_default();
+        let entity_path = if let Some(ref sub_queue) = options.sub_queue {
+            format!("{}{}", base_path, sub_queue.as_path_suffix())
+        } else {
+            base_path
+        };
+
+        self.authorize_path(&entity_path).await?;
+
+        Receiver::new(
+            self.connection.clone(),
+            entity_path,
+            None,
+            options.receive_mode,
+            self.options.clone(),
+        )
+        .await
+    }
+
+    /// Gets the fully qualified namespace.
+    pub fn fully_qualified_namespace(&self) -> &str {
+        &self.namespace
+    }
+
+    /// Authorizes access to a Service Bus entity path using the configured credential.
+    /// This method is used internally by senders and receivers when authentication is required.
```

Comment:
A slight nit: The authorizer will be needed for both SAS token and bearer token authorization - all tokens have an expiration time.